docs/asset_fields_overview.md

```diff
@@ -0,0 +1,44 @@
+## Asset fields overview
+
+Asset Manager stores a a number of fields whose values work together to enforce how the asset is served.
```

Comment:
```suggestion
## Asset state machine overview

Asset Manager stores a a number of fields whose values work together to represent the state of an asset.
```
I know the Asset has a single `state` field, but in reality its state is comprised of all of these fields, as you correctly identify. I think being clear that what we're really documenting here is the state machine is helpful